### PR TITLE
add COMPOSER_MR_TITLE_PREFIX variable for user-configurable MR titles

### PR DESCRIPTION
**Pull request by macbookandrew**
Purpose for this change: we’d like to customize the MR title format as some of our clients see those and we’d like them to be a bit more generic.

This is the first time I’ve tinkered with Go; does everything look ok? I wrote it based on the other code in the repo and didn’t test it as it seems fairly straightforward and I’m not familiar enough with Go to know about building/compiling/running/etc.

### CODE REVIEW

File: app/config.go [hunk 1]
```diff
@@ -76,6 +79,8 @@ func BuildConfig() {
 
 	Config.GitCommitTitle = envString("COMPOSER_MR_COMMIT_TITLE", Config.GitCommitTitle)
 
+	Config.MRTitlePrefix = envString("COMPOSER_MR_TITLE_PREFIX", "Composer update: ")
```
Comment (axllent) [inline app/config.go]: This should be 
```go
Config.MRTitlePrefix = envString("COMPOSER_MR_TITLE_PREFIX", Config.MRTitlePrefix)
```

In addition to this, `cmd/root.go` should get a new flag withim the `func init()` to set the default for that config variable, maybe something like:
```go
rootCmd.Flags().StringVarP(&app.Config.MRTitlePrefix, "title-prefix", "p", "Composer update:", "The merge request title prefix")
```
There is also another change I'll comment on shortly.

File: cmd/root.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ Documentation:
 			pkgs = "packages"
 		}
 
-		mrTitle := fmt.Sprintf("Composer update: %d %s", len(diff.Packages), pkgs)
+		mrTitle := fmt.Sprintf(app.Config.MRTitlePrefix + "%d %s", len(diff.Packages), pkgs)
```
Comment (axllent) [inline cmd/root.go]:
```go
mrTitle := fmt.Sprintf("%s %d %s", app.Config.MRTitlePrefix, len(diff.Packages), pkgs)
```

**Comment by axllent** [comment]
Thanks for your contribution, not a bad effort for someone who doesn't understand Go! I've made a few suggestions just to tidy things up a little.

One point to note is the current `Composer update: ` prefix is used not only to prefix new MRs, but also to delete old MRs (if they haven't been merged yet). Just take note with what you call these - if any MRs from others exist with the same prefix then those too will be deleted (closed).

**Comment by macbookandrew** [comment]
> Thanks for your contribution, not a bad effort for someone who doesn't understand Go! I've made a few suggestions just to tidy things up a little.

Thanks! Those make sense.

> One point to note is the current Composer update:  prefix is used not only to prefix new MRs, but also to delete old MRs (if they haven't been merged yet). Just take note with what you call these - if any MRs from others exist with the same prefix then those too will be deleted (closed).

Great point

**Comment by axllent** [comment]
> > One point to note is the current Composer update:  prefix is used not only to prefix new MRs, but also to delete old MRs (if they haven't been merged yet). Just take note with what you call these - if any MRs from others exist with the same prefix then those too will be deleted (closed).
> 
> Great point

Before I merge this in, I wanted to check if that going to be an issue for you?

**Comment by macbookandrew** [comment]
No, we can manually update old MR titles to match or handle them some other way.

**Comment by axllent** [comment]
Thanks for your work @macbookandrew - this has now been merged and released to Docker hub. Please confirm it works as you expect? Thanks!

**Comment by macbookandrew** [comment]
Thanks; it worked great!

**Pull request merged**